GameDesign/Assets/Scripts/TaserManager.cs

```diff
@@ -2,53 +2,84 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class TaserManager : MonoBehaviour{
+public class TaserManager
+{
+    private static TaserManager instance;
 
-    //Taser variables
-    [SerializeField] private int taserBattery;
-    [SerializeField] private int taserCost = 20;
-    [SerializeField] private int taserMaxBattery = 100;
-    [SerializeField] private bool trigerIsFree = true;
-    [SerializeField] private int chargePerSecond = 2;
-
-    private MeshRenderer _rendered;
-    private Color originalColor;
-    
-
-    // Start is called before the first frame update
-    private void Start(){
-        _rendered = GetComponent<MeshRenderer>();
-        originalColor = Color.blue;
-        _rendered.material.color = originalColor;
-        InvokeRepeating("ChargeBattery", 1.0f, 1.0f);
-    }
-    
-    private void OnMouseOver(){
-        if (Input.GetMouseButton(1) && trigerIsFree)
+    public static TaserManager Instance
+    {
+        get
         {
-            trigerIsFree = false;
-            if (taserBattery>=taserCost )
+            if (instance == null)
             {
-                taserBattery = taserBattery - taserCost;
-                _rendered.material.color = Color.yellow;
+                instance = new TaserManager();
             }
-            Debug.Log($"Decreased battery to: {taserBattery}");
-        }else
-        {
-            _rendered.material.color = originalColor;
+            return instance;
+        }
+    }
+
+    //Taser variables
+    public int taserBattery;
+    public int taserCost = 20;
+    public int taserMaxBattery = 100;
+    public bool triggerIsFree = true;
+    public int chargePerSecond = 2;
+
+    private TaserManager()
+    {
+        taserBattery = taserMaxBattery;
+    }
+
+    public void taserTrigger(){
```

Comment:
If we have these methods, why don't we call them in the `CustomerMonoBehaviour` instead of applying the logic there? (Lines 43-45 in `CustomerMonoBehaviour`

Reply:
I never used them in the end. Just removed them.

Reply:
My point is that we should use these, as the responsibility of handling the Taser behaviour is of the TaserManager, not the customer

Reply:
Why. That's why I use an instance of the taserManager. What would be the difference in calling a function to calling attributes of the taser in the customer object.

Reply:
done